bioimageio/spec/shared/io.py

```diff
@@ -0,0 +1,336 @@
+        raw_model: RawModelNode = io_cls.load_raw_model_from_dict(data)
+        assert isinstance(raw_model, io_cls.raw_nodes.Model)
+
+        return raw_model
+
+    @classmethod
+    def serialize_raw_model_to_dict(cls, raw_model: RawModelNode) -> dict:
+        io_cls = cls.get_matching_io_class(raw_model.format_version, "serialize")
+        serialized = io_cls.schema.Model().dump(raw_model)
+        assert isinstance(serialized, dict)
+        return serialized
+
+    @classmethod
+    def save_raw_model(cls, raw_model: RawModelNode, path: pathlib.Path):
+        warnings.warn("only saving serialized rdf, no associated resources.")
+        if path.suffix != ".yaml":
+            warnings.warn("saving with '.yaml' suffix is strongly encouraged.")
+
+        serialized = cls.serialize_raw_model_to_dict(raw_model)
+        yaml.dump(serialized, path)
+
+    @classmethod
+    def serialize_raw_model(cls, raw_model: Union[dict, RawModelNode]) -> str:
+        if not isinstance(raw_model, dict):
+            raw_model = cls.serialize_raw_model_to_dict(raw_model)
+
+        with StringIO() as stream:
+            yaml.dump(raw_model, stream)
+            return stream.getvalue()
+
+    @classmethod
+    def load_model(
+        cls,
+        source: Union[RawModelNode, os.PathLike, str, dict],
+        root_path: Optional[os.PathLike] = None,
+        update_to_current_format: bool = True,
+    ):
+        raw_model, root_path = cls.ensure_raw_model(source, root_path, update_to_current_format)
+
+        matching_cls = cls.get_matching_io_class(raw_model.format_version, "load")
+
+        model: ModelNode = resolve_raw_node_to_node(
+            raw_node=raw_model, root_path=pathlib.Path(root_path), nodes_module=matching_cls.nodes
+        )
+        assert isinstance(model, matching_cls.nodes.Model)
+
+        return model
+
+    @classmethod
+    def export_package(
+        cls,
+        source: Union[RawModelNode, os.PathLike, str, dict],
+        root_path: Optional[os.PathLike] = None,
+        update_to_current_format: bool = False,
+        weights_formats_priorities: Optional[Sequence[current_spec.raw_nodes.WeightsFormat]] = None,
+    ) -> pathlib.Path:
+        """
+        weights_formats_priorities: If given only the first weights format present in the model is included.
+                                    If none of the prioritized weights formats is found all are included.
+        """
+        if isinstance(source, str) and source.startswith("http"):
+            raise ValueError("Cannot export package for remote source")
+
+        raw_model, root_path = cls.ensure_raw_model(source, root_path, update_to_current_format)
+        io_cls = cls.get_matching_io_class(raw_model.format_version, "export")
+        package_path = io_cls._make_package_wo_conversions(
+            raw_model, root_path, weights_formats_priorities=weights_formats_priorities
+        )
+        return package_path
```

Comment:
> 1) used to produce a zip package directly

this class method is exposed as `export_package` for every subversion (0.1, 0.3) and returns a path to the zipped package in the systems <tmp>/bioimageio_cache folder